Code quality : overrides.

ssj17vegeta opened the pull request:
## 🍰 Pullrequest
Treated a number of issues reported by CppCheck (error type "missingOverride").

- Added "virtual" and "override" specifiers when needed.
- Removed unneeded virtual destructors that were empty.
- Removed duplicate member declarations on child classes/structs (members already existed in parent classes/structs).

ssj17vegeta committed:
Code quality : overrides.

Treated a number of issues reported by CppCheck (error type "missingOverride").

- Added "virtual" and "override" specifiers when needed.
- Removed unneeded virtual destructors that were empty.
- Removed duplicate member declarations on child classes/structs (members already existed in parent classes/structs).

ssj17vegeta commented on src/game/Movement/spline.h:
```diff
@@ -123,7 +123,7 @@ namespace Movement
             }
 
-            void clear();
+            virtual void clear();
```
Based on my previous comment, feel free to tell me whatever solution you prefer. If those two virtuals have to go, I'll edit the pull request.

evil-at-wow replied:
A quick grep seems to suggest we're not relying on SplineBase except for one enum, so I would personally remove the virtual for now. If somewhere we're using splines using a SplineBase pointer, chances are high we got other issues already (like no virtual destructor if we destroy them via a pointer to SplineBase, perhaps even slicing, and so on).

But that still leaves the warning of course - GCC/clang can also warn about hiding functions. So I think I'm going to suggest doing something like for now (as part of this PR):

```
...
using SplineBase::clear;
void clear();
...
```

That should fix the warning too, without (more) adverse effects, even though it doesn't really feel right, design wise. But I don't know the details about the spline related coded, so I'm not sure what the best option would be. Looking at the code, it almost feels like SplineBase is really meant to be a base class, but Spline is currently the only derived class with no indication that should ever change, so then again it doesn't. Git history suggests it was like this from the beginning of the Spline introduction, so - again without knowing any details about the why it was done this way - it feels like we just need a template class Spline and "merge" the two existing classes, for a better design that avoids all the potential issues we have now.

ssj17vegeta committed:
Partially revert adding "virtual" on two classes.

- Reverting the adding on virtual on two classes (RegularGrid and Spline) for performance purposes. Also removed the override specifiers previously added since it won't compiled otherwise. A rework might be needed if these classes aren't meant to be inherited.

- Added a missing include in PathFinder.cpp that prevented compiling when using DPCH=0.
